### PR TITLE
[DPM-227] use string instead of uint64 in requests

### PR DESCRIPTION
**Pull request by algys**

### CODE REVIEW

File: server/api/handlers/account_info.go [hunk 1]
```diff
@@ -2,15 +2,37 @@ package handlers
...
 )
 
+type PlayerInfoResponse struct {
+	Balance          eos.Asset         `json:"balance"`
+	ActivePermission eos.Authority     `json:"activePermission"`
+	OwnerPermission  eos.Authority     `json:"ownerPermission"`
+	LinkedCasinos    []*CasinoResponse `json:"linkedCasinos"`
+}
+
+func toPlayerInfoResponse(p *models.PlayerInfo) *PlayerInfoResponse {
```
Comment (Garik-) [inline server/api/handlers/account_info.go:17]: I would do so. in my opinion so beautiful
``` func (p *models.PlayerInfo) toPlayerInfoResponse() *PlayerInfoResponse```
```return player.toPlayerInfoResponse(), nil```

File: server/api/handlers/fetch_casinos.go [hunk 2]
```diff
@@ -2,14 +2,35 @@ package handlers
...
+	"strconv"
 )
 
+type CasinoResponse struct {
+	Id       string `json:"id"`
+	Contract string `json:"contract"`
+	Paused   bool   `json:"paused"`
+}
+
+func toCasinoResponse(c *models.Casino) *CasinoResponse {
```
Comment (Garik-) [inline server/api/handlers/fetch_casinos.go:16]: See comment above

File: server/api/handlers/fetch_casinos.go [hunk 3]
```diff
@@ -2,14 +2,35 @@ package handlers
...
+}
+
 func ProcessFetchCasinosRequest(context context.Context, req *ws_interface.ApiRequest) (interface{}, *ws_interface.HandlerError) {
 	casinos, err := req.Repos.Contracts.AllCasinos(context)
 	if err != nil {
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
-	return casinos, nil
+	var response []*CasinoResponse
```
Comment (Garik-) [inline server/api/handlers/fetch_casinos.go:30]: You already know the size of the data. It’s better to immediately allocate the right size memory.
`response := make([]*CasinoResponse, len(casinos))`

File: server/api/handlers/fetch_games.go [hunk 4]
```diff
@@ -2,14 +2,40 @@ package handlers
...
+}
+
 func ProcessFetchGamesRequest(context context.Context, req *ws_interface.ApiRequest) (interface{}, *ws_interface.HandlerError) {
 	games, err := req.Repos.Contracts.AllGames(context)
 	if err != nil {
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
-	return games, nil
+	var response []*GameResponse
```
Comment (Garik-) [inline server/api/handlers/fetch_games.go:35]: `response := make([]*GameResponse, len(games))`

File: server/api/handlers/fetch_games.go [hunk 5]
```diff
@@ -2,14 +2,40 @@ package handlers
...
 
+type GameResponse struct {
+	Id        string          `json:"id"`
+	Contract  string          `json:"contract"`
+	ParamsCnt uint16          `json:"paramsCnt"`
+	Paused    int             `json:"paused"`
+	Meta      json.RawMessage `json:"meta"`
+}
+
+func toGameResponse(g *models.Game) *GameResponse {
```
Comment (Garik-) [inline server/api/handlers/fetch_games.go:19]: See comment above

File: server/api/handlers/fetch_games_in_casino.go [hunk 6]
```diff
@@ -3,12 +3,42 @@ package handlers
...
+	Params []GameParamResponse `json:"params"`
+}
+
+func toCasinoGameResponse(g *models.CasinoGame) *CasinoGameResponse {
+	ret := &CasinoGameResponse{
+		Id:     strconv.FormatUint(g.Id, 10),
+		Paused: g.Paused,
+	}
+
+	for _, param := range g.Params {
```
Comment (Garik-) [inline server/api/handlers/fetch_games_in_casino.go:34]: Here you also already know what size the array will be.
`make(..., len()). for i, param: = range.  ret.Prams[i] =`

File: server/api/handlers/account_info.go [hunk 7]
```diff
@@ -2,15 +2,37 @@ package handlers
...
+	ret := &PlayerInfoResponse{
+		Balance:          p.Balance,
+		ActivePermission: p.ActivePermission,
+		OwnerPermission:  p.OwnerPermission,
+	}
+	for _, casino := range p.LinkedCasinos {
```
Comment (Garik-) [inline server/api/handlers/account_info.go:23]: Size is known in advance; it is better to allocate memory of the right size immediately

File: server/api/handlers/fetch_session_updates.go [hunk 8]
```diff
@@ -35,5 +55,10 @@ func ProcessFetchSessionUpdatesRequest(context context.Context, req *ws_interfac
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
-	return gameSessionUpdates, nil
+	var response []*SessionUpdateResponse
+	for _, su := range gameSessionUpdates {
```
Comment (Garik-) [inline server/api/handlers/fetch_session_updates.go:59]: make(..., len)

File: server/api/handlers/fetch_sessions.go [hunk 9]
```diff
@@ -11,5 +11,10 @@ func ProcessFetchSessionsRequest(context context.Context, req *ws_interface.ApiR
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
-	return gameSessions, nil
+	var response []*GameSessionResponse
+	for _, session := range gameSessions {
```
Comment (Garik-) [inline server/api/handlers/fetch_sessions.go:15]: make(..., len)

File: server/api/handlers/game_action.go [hunk 10]
```diff
@@ -37,7 +38,12 @@ func ProcessGameActionRequest(context context.Context, req *ws_interface.ApiRequ
 		return nil, ws_interface.NewHandlerError(ws_interface.UnauthorizedError, errors.New("attempt to play not own session"))
 	}
 
-	err = req.UseCases.GameSession.GameAction(context, payload.SessionId, payload.ActionType, payload.Params)
+	var params []uint64
+	for _, param := range payload.Params {
```
Comment (Garik-) [inline server/api/handlers/game_action.go:42]: make(..., len)

File: server/api/handlers/account_info.go (hunk 1 above)
Comment (algys) [reply server/api/handlers/account_info.go:17]: I cant specify method for external type. In that case `models.PlayerInfo` is external.
It's possible only if we would define that methods in models package, but then we get abstraction leak.

File: server/api/handlers/fetch_casinos.go (hunk 2 above)
Comment (algys) [reply server/api/handlers/fetch_casinos.go:16]: see answer above)

File: server/api/handlers/fetch_casinos.go (hunk 3 above)
Comment (algys) [reply server/api/handlers/fetch_casinos.go]: fixed

File: server/api/handlers/new_game.go [hunk 11]
```diff
@@ -58,18 +59,23 @@ func ProcessNewGameRequest(context context.Context, req *ws_interface.ApiRequest
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
+	var actionParams []uint64
+	for _, param := range payload.ActionParams {
```
Comment (Garik-) [inline server/api/handlers/new_game.go:63]: make(..., len) ?

File: server/api/handlers/game_action.go (hunk 10 above)
Comment (algys) [reply server/api/handlers/game_action.go]: fixed

**Pull request merged**